### DISCUSSION TITLE
Deletion of auto-generated albums

### DISCUSSION THREAD

**Discussion by kvalev** [photoprism/photoprism#1725] [Ideas]
Right now the deletion policy for auto-generated albums is inconsistent - some can be deleted, others not. What do you think would be the correct behavior?

**Comment by lastzero** [comment]
As far as I can see, only folder albums can't be deleted manually. This makes sense in that they are "coupled" to folders on your disk. So you can delete them there if you don't like them?

**Reply by kvalev** [reply]
Also `month` albums cannot be deleted, which is what makes the discrepancy weird. Anyway my suggestion would be to prohibit deleting all auto-generated albums (`state` and `moments`) and instead photoprism should decide whether to delete them during indexing (if they become empty).

**Reply by lastzero** [reply]
The visibility of months and folders is handled automatically, depending on whether there are folders and whether there are images taken in a particular month. The calendar page was intended as a "replacement" for a timeline view, which some users may  not see or use that way. For me personally, it's easier than fumbling around with a scroll bar. Now imagine random months were "missing" in a timeline view because somebody deleted them days or years ago. Wouldn't that be super strange?

**Reply by kvalev** [reply]
No argument there, my point is actually to prohibit deletion of `state` and `moments` albums as well using the same logic - it would be super strange to have missing states/moments because someone deleted them.

**Reply by lastzero** [reply]
Right, it's a bit of a workaround for States as they don't have photo counts yet. So we can't hide them when the count is zero. AFAIK there is no index on state names, so keeping counts updated all time requires too many resources. An index would be wasteful too if only used for this.

You could always delete moments as they are not connected with each other in a way months are in a calender (or wheels on a car if you like). When PhotoPrism creates a moment of you and your ex partner, you don't have to keep it. Trust me, users want this when the moment is there.

**Reply by lastzero** [reply]
FYI: Visibility of States & Months is toggled automatically with the latest changes. Moments should stay deletable in case you have an issue (whatsoever) with one of the suggestions. Starting a Development Preview build now. Release Notes will be updated soon too.

**Reply by kvalev** [reply]
What do you mean by "toggled"? States & Months have always been visible.

**Reply by lastzero** [reply]
Now they will be hidden when empty if you give it a few minutes.
